x-pack/plugins/security/server/plugin.ts

```diff
@@ -267,8 +295,19 @@ export class Plugin {
 
         return this.authenticationStart;
       },
+      getAnonymousAccessService: () => {
+        if (!this.anonymousAccessStart) {
+          throw new Error('AnonymousAccess service is not started!');
+        }
+
+        return this.anonymousAccessStart;
+      },
     });
 
+    core.capabilities.registerProvider(() => ({
```

Comment:
I'm also hesitant about this. UI Capabilities were designed to support FC, but that doesn't mean they can't be used for other purposes. That said, I'm not convinced this is the right tool for the job.


> The main motivator is to avoid unnecessary call to the Kibana server on every page refresh just to know that anonymous access isn't enabled (yeah, I'm missing good old injected vars :see_no_evil: )

I miss injected vars too! If the network call is the primary motivation, then perhaps we can tweak the `security_oss` plugin a bit. The OSS plugin [makes a call](https://github.com/azasypkin/kibana/blame/ad82f4ca8748ec9a29b76de87af394f9506e6f47/src/plugins/security_oss/public/insecure_cluster_service/insecure_cluster_service.tsx#L101-L108) to determine if the "insecure cluster warning" should be shown.

If we wanted to, we could introduce an `/internal/security/setup` (or similar) call, which can return information for both the "insecure cluster warning" and the state of anonymous access. That would keep the number of network calls the same as it is today, without changing the way that UI Capabilities are used. What do you think?

Reply:
> If we wanted to, we could introduce an /internal/security/setup (or similar) call, which can return information for both the "insecure cluster warning" and the state of anonymous access. That would keep the number of network calls the same as it is today, without changing the way that UI Capabilities are used. What do you think?

Yeah, I like that! I'll try to do that.